core/Modules/MxKarma/MxKarma.php

```diff
@@ -182,6 +183,18 @@ public static function playerConnect(Player $player)
         Template::show($player, 'MxKarma.mx-karma', ['total' => self::$votesTotal, 'average' => self::$voteAverage]);
     }
 
+    public static function playerPb(Player $player, int $score)
+    {
+        if ($score == 0) {
+            return;
+        }
+        $map = MapController::getCurrentMap();
+        $vote = DB::table('mx-karma')->where('Player', '=', $player->id)->where('map', '=', $map->id)->first();
```

Comment:
To speed things up you can check with `->exists()`

Reply:
Thanks, you're right. It should also actually fix a logic error with that part of the code.